content/lessons/chapter-2/mining-1.tsx

```diff
@@ -90,38 +98,56 @@ export default function Mining1({ lang }) {
   }, [ramdomNonce])
 
   const transactionStep = () => {
-    setStep(5)
+    setShowText(false)
+    setTimeout(() => {
+      setStep(5)
+    }, 300)
```

Comment:
Lets create a util function:
```ts
function sleep(timeout: number) {
  return new Promise(resolve => {
    const t = setTimeout(() => {
      resolve()
      clearTimeout(t)
    }, timeout)
  })
}
```

Then we can use it like this by making these function async:
```ts
const transactionStep = async () => {
  setShowText(false)
  setNonceHighlight(false)
  setHashPowerHighlight(false)
  setTransactionsConfirmedHighlight(true)
  await sleep(300)
  setStep(5)
})